xbmc/addons/kodi-dev-kit/include/kodi/tools/Thread.h

```diff
@@ -0,0 +1,398 @@
+/// };
+///
+/// ADDON_STATUS CTestAddon::Create()
+/// {
+///   kodi::Log(ADDON_LOG_INFO, "Starting thread");
+///   CreateThread();
+///
+///   Sleep(4000);
+///
+///   kodi::Log(ADDON_LOG_INFO, "Stopping thread");
+///   // This added as example and also becomes stopped by class destructor
+///   StopThread();
+///
+///   return ADDON_STATUS_OK;
+/// }
+///
+/// void CTestAddon::Process()
+/// {
+///   kodi::Log(ADDON_LOG_INFO, "Thread started");
+///
+///   while (!m_threadStop)
+///   {
+///     kodi::Log(ADDON_LOG_INFO, "Hello World");
+///     Sleep(1000);
+///   }
+///
+///   kodi::Log(ADDON_LOG_INFO, "Thread ended");
+/// }
+///
+/// ADDONCREATOR(CTestAddon)
+/// ~~~~~~~~~~~~~
+///
+///@{
+class CThread
+{
+public:
+  //============================================================================
+  /// @ingroup cpp_kodi_tools_CThread
+  /// @brief Class constructor.
+  ///
+  CThread() : m_threadStop(false) {}
+  //----------------------------------------------------------------------------
+
+  //============================================================================
+  /// @ingroup cpp_kodi_tools_CThread
+  /// @brief Class destructor.
+  ///
+  virtual ~CThread()
+  {
+    StopThread();
+    if (m_thread != nullptr)
+    {
+      m_thread->detach();
+      delete m_thread;
+    }
+  }
+  //----------------------------------------------------------------------------
+
+  //============================================================================
+  /// @ingroup cpp_kodi_tools_CThread
+  /// @brief Check auto delete is enabled on this thread class.
+  ///
+  /// @return true if auto delete is used, false otherwise
+  ///
+  bool IsAutoDelete() const { return m_autoDelete; }
+  //----------------------------------------------------------------------------
+
+  //============================================================================
+  /// @ingroup cpp_kodi_tools_CThread
+  /// @brief Check caller is on this running thread.
+  ///
+  /// @return true if called from thread inside the class, false if from another
+  ///         thread
+  ///
+  bool IsCurrentThread() const { return m_threadId == std::this_thread::get_id(); }
+  //----------------------------------------------------------------------------
+
+  //============================================================================
+  /// @ingroup cpp_kodi_tools_CThread
+  /// @brief Check thread inside this class is running and active.
+  ///
+  /// @note This function should be used from outside and not within process to
+  /// check thread is active. Use use atomic bool @ref m_threadStop for this.
+  ///
+  /// @return true if running, false if not
+  ///
+  bool IsRunning() const
+  {
+    if (m_thread != nullptr)
+    {
+      // it's possible that the thread exited on it's own without a call to StopThread. If so then
+      // the promise should be fulfilled.
+      std::future_status stat = m_future.wait_for(std::chrono::milliseconds(0));
```

Comment:
`#include <chrono>`is missing.

Reply:
Thanks a lot for see. Have added the missing header.